src/passes/userDefinedTypesConverter.ts

```diff
@@ -100,3 +94,43 @@ export class UserDefinedTypesConverter extends ASTMapper {
     return ast;
   }
 }
+
+function replaceUserDefinedType(type: TypeNode): TypeNode {
+  if (type instanceof ArrayType) {
+    return new ArrayType(replaceUserDefinedType(type.elementT), type.size, type.src);
+  } else if (type instanceof FunctionType) {
+    return new FunctionType(
+      type.name,
+      type.parameters.map(replaceUserDefinedType),
+      type.returns.map(replaceUserDefinedType),
+      type.visibility,
+      type.mutability,
+      type.src,
+    );
+  } else if (type instanceof MappingType) {
+    return new MappingType(
+      replaceUserDefinedType(type.keyType),
+      replaceUserDefinedType(type.valueType),
+      type.src,
+    );
+  } else if (type instanceof PointerType) {
+    return new PointerType(replaceUserDefinedType(type.to), type.location, type.kind, type.src);
+  } else if (type instanceof TupleType) {
+    return new TupleType(type.elements.map(replaceUserDefinedType), type.src);
+  } else if (type instanceof TypeNameType) {
+    return new TypeNameType(replaceUserDefinedType(type.type), type.src);
+  } else if (type instanceof UserDefinedType) {
+    if (type.definition instanceof UserDefinedValueTypeDefinition) {
+      return typeNameToTypeNode(
+        new ElementaryTypeName(
```

Comment:
What's the motivation behind creating a new node here? I don't see what this covers that `typeNameToTypeNode(type.definition.underlyingType)` doesn't

Reply:
Ah my bad. I didn't think of this. I had the misconception that I had to convert the underlying typestring into an `ElementaryTypeName` node and then call the function on it. Not realising that `type.definition.underlyingType` already has the required node.